### PR TITLE
Move download functions from `__init__` to its own module

### LINKED ISSUE #372: Move code from __init__.py to its own module
### 📝 Description of the feature

There is a lot of code to download individual files in this `__init__.py` file: https://github.com/pyansys/pyprimemesh/blob/main/src/ansys/meshing/prime/examples/__init__.py

I personally think that these functions being here causes confusion, as I would never look for them there, so I would move them to their own module inside that folder.

### 💡 Steps for implementing the feature

- Move the code to its own module
- Manage the import of the new module on the `__init__.py` file so the change is transparent for the rest of the modules.

### 🔗 Useful links and references

_No response_

### PR DESCRIPTION
**Pull request by AlejandroFernandezLuces**
## Explanation
I've found that there was a lot of download functions in `examples/__init__.py`. There shouldn't be any logic implemented in `__init__` files, we should only allow import simplifications, configuration variables and code that should be run when importing the module.

I've moved the functions in this file to a more appropriate file. Also, I renamed the files to improve readability.

Closes #372

### CODE REVIEW

File: src/ansys/meshing/prime/examples/examples.py [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+        Destination for the file to be downloaded.
+        If nothing is provided, the default path in app data is used
+    force: bool
+        If true, the file is always downloaded.
+        If false, an existing file in the cache may be re-used.
+
+    Returns
+    -------
+    str
+        Local path to the downloaded file
```
Comment (MaxJPRey) [inline src/ansys/meshing/prime/examples/examples.py:43]:
```suggestion
        Local path to the downloaded file.
```

File: src/ansys/meshing/prime/examples/examples.py [hunk 2]
```diff
@@ -0,0 +1,487 @@
...
+) -> Union[str, os.PathLike]:
+    """Download PMDAT file for the mixing elbow example
+
+    Parameters
+    ----------
+    destination: Optional[str]
+        Destination for the file to be downloaded.
+        If nothing is provided, the default path in app data is used
+    force: bool
+        If true, the file is always downloaded.
```
Comment (MaxJPRey) [inline src/ansys/meshing/prime/examples/examples.py:37]:
```suggestion
        If ``True``, the file is always downloaded.
```

File: src/ansys/meshing/prime/examples/examples.py [hunk 3]
```diff
@@ -0,0 +1,487 @@
...
+        If false, an existing file in the cache may be re-used.
```
Comment (MaxJPRey) [inline src/ansys/meshing/prime/examples/examples.py:38]:
```suggestion
        Otherwise, an existing file in the cache may be re-used.
```

File: src/ansys/meshing/prime/examples/examples.py [hunk 4]
```diff
@@ -0,0 +1,487 @@
...
+
+    Returns
+    -------
+    str
+        Local path to the downloaded file
+
+    Raises
+    ------
+    ValueError
+        When the provided destination path does not exist on file
```
Comment (MaxJPRey) [inline src/ansys/meshing/prime/examples/examples.py:48]:
```suggestion
        Exception is raised when the provided destination path does not exist.
```

File: src/ansys/meshing/prime/examples/examples.py [hunk 5]
```diff
@@ -0,0 +1,487 @@
...
+    >>>     print(model)
+
+    """
+    if destination is not None and not os.path.isdir(destination):
+        raise ValueError('destination directory provided does not exist')
+    file = download_file(
+        'mixing_elbow.pmdat', 'pyprimemesh', 'mixing_elbow', destination=destination, force=force
+    )
+    _DOWNLOADS.append(file)
+    return file
```
Comment (MaxJPRey) [inline src/ansys/meshing/prime/examples/examples.py:68]: @AlejandroFernandezLuces It seems that this code is duplicated many times -except the file path-. Could we create a common function to replace it?

File: src/ansys/meshing/prime/examples/examples.py [hunk 6]
```diff
@@ -5,450 +5,38 @@
...
 
     Parameters
     ----------
+    file_name : str
+        Name of the file.
+    file_extension : str
+        Extension of the file.
     destination: Optional[str]
         Destination for the file to be downloaded.
         If nothing is provided, the default path in app data is used
```
Comment (MaxJPRey) [inline src/ansys/meshing/prime/examples/examples.py:36]:
```suggestion
       The default is ``None`` in which case the default path in app data is used.
```

**Review by MaxJPRey — APPROVED** [review]

File: examples/gallery/bracket_scaffold.py [hunk 7]
```diff
@@ -57,7 +57,7 @@
 
 # . Create part per CAD model for topology based connection.
 
-bracket_file = prime.examples.download_bracket_fmd()
```
Comment (ninad-kamat) [inline examples/gallery/bracket_scaffold.py:60]: One of the reasons to have specific functions for specific files is that they are documented as part of our API reference. With this, how does a new user now these magic strings that you are supposed to pass to `get_file`?

Comment (AlejandroFernandezLuces) [reply examples/gallery/bracket_scaffold.py:59]: Good point! I've used an enum instead of magic strings, let me know what do you think.

File: examples/gallery/bracket_scaffold.py [hunk 8]
```diff
@@ -54,9 +55,9 @@
 # Create part per CAD model for topology based connection.
 
 # For Windows OS users scdoc is also available:
-# bracket_file = prime.examples.download_bracket_scdoc()
+# bracket_file = prime.examples.download_bracket_scdoc(Examples.BRACKET_SCDOC)
```
Comment (waltersma) [inline examples/gallery/bracket_scaffold.py:58]: @AlejandroFernandezLuces 
```suggestion
# bracket_file = prime.examples.get_file(Examples.BRACKET_SCDOC)
```

File: examples/gallery/lucid_mixing_elbow.py [hunk 9]
```diff
@@ -55,10 +56,12 @@
...
 # Create face zones from labels imported from geometry for use in Fluent solver.
 
+
 # For Windows OS users scdoc is also available:
-# mixing_elbow = prime.examples.download_elbow_scdoc()
+# mixing_elbow = prime.examples.get_file(Examples.ELBOW_SCDOC)
 
-mixing_elbow = prime.examples.download_elbow_fmd()
+# . Create face zones from labels imported from geometry for use in Fluent solver.
+mixing_elbow = prime.examples.get_file(Examples.ELBOW_PMDAT)
```
Comment (waltersma) [inline examples/gallery/lucid_mixing_elbow.py:64]: @AlejandroFernandezLuces any reason why we have changed the input format to pmdat on these files?

Comment (waltersma) [inline examples/gallery/lucid_mixing_elbow.py:64]:
```suggestion
mixing_elbow = prime.examples.get_file(Examples.ELBOW_FMD)
```

File: examples/gallery/lucid_pipe_tee.py [hunk 10]
```diff
@@ -66,8 +67,7 @@
 # . Read and display the geometry file.
 
 # The file contains several unmeshed parts as you would get after you imported from CAD.
-
-file_name = prime.examples.download_pipe_tee_fmd()
+file_name = prime.examples.get_file(Examples.PIPE_TEE_PMDAT)
```
Comment (waltersma) [inline examples/gallery/lucid_pipe_tee.py:70]: @AlejandroFernandezLuces 
```suggestion
file_name = prime.examples.get_file(Examples.PIPE_TEE_FMD)
```

File: examples/gallery/lucid_toy_car.py [hunk 11]
```diff
@@ -57,10 +58,11 @@
 # Download the toy car geometry file (.fmd file exported by SpaceClaim).
 # Import geometry and display everything except tunnel.
 
+
 # For Windows OS users scdoc is also available:
-# toy_car = prime.examples.download_toy_car_scdoc()
+# toy_car = prime.examples.get_file(Examples.TOY_CAR_SCDOC)
 
-toy_car = prime.examples.download_toy_car_fmd()
+toy_car = prime.examples.get_file(Examples.TOY_CAR_PMDAT)
```
Comment (waltersma) [inline examples/gallery/lucid_toy_car.py:65]: @AlejandroFernandezLuces 
```suggestion
toy_car = prime.examples.get_file(Examples.TOY_CAR_FMD)
```

File: examples/misc/example_template.py [hunk 12]
```diff
@@ -44,8 +45,8 @@
 
 # Your code goes here...
 mesh_util = prime.lucid.Mesh(model=model)
-example_file = prime.examples.download_elbow_fmd()
-mesh_util.read(example_file)
+mixing_elbow = prime.examples.get_file(Examples.ELBOW_PMDAT)
```
Comment (waltersma) [inline examples/misc/example_template.py:48]: @AlejandroFernandezLuces 
```suggestion
mixing_elbow = prime.examples.get_file(Examples.ELBOW_FMD)
```

File: examples/gallery/lucid_mixing_elbow.py (hunk 9 above)
Comment (AlejandroFernandezLuces) [reply examples/gallery/lucid_mixing_elbow.py]: No reason, must have been a confusion.

**Review by ninad-kamat — APPROVED** [review]

**Comment by AlejandroFernandezLuces** [comment]
This is ready to merge. I need a new approving review, as I changed the download examples interface to comply with how it was done before this PR.

File: examples/gallery/lucid_mixing_elbow.py [hunk 13]
```diff
@@ -55,9 +55,11 @@
 # Import geometry.
 # Create face zones from labels imported from geometry for use in Fluent solver.
 
+
 # For Windows OS users scdoc is also available:
 # mixing_elbow = prime.examples.download_elbow_scdoc()
 
+# . Create face zones from labels imported from geometry for use in Fluent solver.
```
Comment (ninad-kamat) [inline examples/gallery/lucid_mixing_elbow.py:62]: @AlejandroFernandezLuces  Was this changes intended or discussed with @waltersma ?

File: examples/gallery/lucid_pipe_tee.py [hunk 14]
```diff
@@ -124,7 +122,7 @@
 # By default, the wrap uses all parts as input and deletes the input
 # geometry after wrapping unless keep_input is set as True.
 
-mesh_util.read(pipe_tee)
+mesh_util.read(file_name)
```
Comment (ninad-kamat) [inline examples/gallery/lucid_pipe_tee.py:125]: @AlejandroFernandezLuces Let us revert to using `pipe_tee` instead of `file_name` to be consistent with other examples.

File: examples/gallery/lucid_mixing_elbow.py (hunk 13 above)
Comment (AlejandroFernandezLuces) [reply examples/gallery/lucid_mixing_elbow.py]: This comment was duplicated from copypasting code. My bad.

**Review by MaxJPRey — APPROVED** [review]
